### ISSUE TITLE
Add a view to UrlDispatcher for iterating over all registered routes.

### ISSUE THREAD

**Issue by asvetlov** [aio-libs/aiohttp#504]
Now the class has support for iterating over named routes only.
We should do the same for all routes (named+unnamed).

**Comment by asvetlov** [comment]
I'd like to have `.named_routes()` as well as `.routes()`.
Any volunteer? The patch is very easy.

**Comment by jashandeep-sohi** [comment]
Should `.named_routes()` return a `dict`-like object or is a `RoutesView` sufficient?

**Comment by asvetlov** [comment]
It should return a read-only `collections.abc.Mapping` I guess.
Do you have another idea?

**Comment by jashandeep-sohi** [comment]
[`types.MappingProxyType`](https://docs.python.org/3/library/types.html#types.MappingProxyType) already implements a read-only interface over a mapping. So I think this should be enough:

```
def named_routes(self):
    return types.MappingProxyType(self._routes)
```

**Comment by asvetlov** [comment]
Wow! You are right.
Would you provide a PR with test and docs?

**Comment by jashandeep-sohi** [comment]
Yes.

**Comment by asvetlov** [comment]
Fixed by #622
